lxd/storage_volumes_snapshot.go

```diff
@@ -640,6 +643,31 @@ func autoCreateCustomVolumeSnapshotsTask(d *Daemon) (task.Func, task.Schedule) {
 				continue
 			}
 
+			// Get list of cluster nodes
+			nodes, err := d.gateway.NodeStore().Get(d.gateway.Context())
+			if err != nil {
+				continue
+			}
+
+			// If there is more than one node (clustering), choose a random node for snapshotting.
+			if len(nodes) > 1 {
+				seed := fmt.Sprintf("%d", v.ID)
+
+				hash := fnv.New64a()
+				_, err = io.WriteString(hash, seed)
+				if err != nil {
+					continue
+				}
+
+				r := rand.New(rand.NewSource(int64(hash.Sum64())))
+				selectedNodeID := r.Int63n(int64(len(nodes))) + 1
```

Comment:
@monstermunchkin this doesn't look quite right.

You're getting a 'random' number between 0 and `len(nodes)`. I would have then expected the returned number to be used to select an index in the `nodes` slice, which then would be used to compare the node ID to.

Otherwise you're relying on node IDs to always be sequential starting from 1 (which I imagine isn't guaranteed to be the case if nodes are removed during a cluster's lifetime).

So it would likely need something like this:

```
// If there are 3 nodes, then we want a random number between 0 and 2, so we can select the nodes[nodeIndex].
nodeIndex := r.Int63n(int64(len(nodes)-1)) 
```

Reply:
This should be fixed now as well.